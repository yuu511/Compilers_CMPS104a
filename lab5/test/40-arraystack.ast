ROOT "<<ROOT>>" 0.0.0 {0} 
FUNCTION "" 6.22.1 {0} void function 
TYPE_ID "" 6.22.1 {0} 
VOID "void" 6.22.1 {0} 
IDENT "fail" 6.22.6 {0} 
PARAM "(" 6.22.11 {0} 
TYPE_ID "" 6.22.12 {1} string variable param lval 
STRING "string" 6.22.12 {0} 
IDENT "expr" 6.22.19 {0} 
TYPE_ID "" 6.22.25 {1} string variable param lval 
STRING "string" 6.22.25 {0} 
IDENT "file" 6.22.32 {0} 
TYPE_ID "" 6.22.38 {1} int variable param lval 
INT "int" 6.22.38 {0} 
IDENT "line" 6.22.42 {0} 
FUNCTION "" 6.24.1 {1} void function 
TYPE_ID "" 6.24.1 {0} 
VOID "void" 6.24.1 {0} 
IDENT "putchr" 6.24.6 {0} 
PARAM "(" 6.24.13 {0} 
TYPE_ID "" 6.24.14 {2} int variable param lval 
INT "int" 6.24.14 {0} 
IDENT "chr" 6.24.18 {0} 
FUNCTION "" 6.25.1 {2} void function 
TYPE_ID "" 6.25.1 {0} 
VOID "void" 6.25.1 {0} 
IDENT "putint" 6.25.6 {0} 
PARAM "(" 6.25.13 {0} 
TYPE_ID "" 6.25.14 {3} int variable param lval 
INT "int" 6.25.14 {0} 
IDENT "num" 6.25.18 {0} 
FUNCTION "" 6.26.1 {3} void function 
TYPE_ID "" 6.26.1 {0} 
VOID "void" 6.26.1 {0} 
IDENT "putstr" 6.26.6 {0} 
PARAM "(" 6.26.13 {0} 
TYPE_ID "" 6.26.14 {4} string variable param lval 
STRING "string" 6.26.14 {0} 
IDENT "str" 6.26.21 {0} 
FUNCTION "" 6.28.1 {4} int function 
TYPE_ID "" 6.28.1 {0} 
INT "int" 6.28.1 {0} 
IDENT "getchr" 6.28.5 {0} 
PARAM "(" 6.28.11 {0} 
FUNCTION "" 6.29.1 {5} string function 
TYPE_ID "" 6.29.1 {0} 
STRING "string" 6.29.1 {0} 
IDENT "getstr" 6.29.8 {0} 
PARAM "(" 6.29.14 {0} 
FUNCTION "" 6.30.1 {6} string function 
TYPE_ID "" 6.30.1 {0} 
STRING "string" 6.30.1 {0} 
IDENT "getln" 6.30.8 {0} 
PARAM "(" 6.30.13 {0} 
STRUCT "struct" 7.7.1 {0} struct stack 
IDENT "stack" 7.7.8 {0} 
TYPE_ID "" 7.8.4 {0} string array field 
ARRAY "array" 7.8.4 {0} 
STRING "string" 7.8.10 {0} 
IDENT "data" 7.8.18 {0} 
TYPE_ID "" 7.9.4 {0} int field 
INT "int" 7.9.4 {0} 
IDENT "size" 7.9.8 {0} 
TYPE_ID "" 7.10.4 {0} int field 
INT "int" 7.10.4 {0} 
IDENT "top" 7.10.8 {0} 
FUNCTION "" 7.13.1 {7} function struct stack 
TYPE_ID "" 7.13.1 {0} 
PTR "ptr" 7.13.1 {0} 
IDENT "stack" 7.13.12 {0} 
IDENT "new_stack" 7.13.19 {0} 
PARAM "(" 7.13.29 {0} 
TYPE_ID "" 7.13.30 {8} int variable param lval 
INT "int" 7.13.30 {0} 
IDENT "size" 7.13.34 {0} 
BLOCK "{" 7.13.40 {8} 
TYPE_ID "" 7.14.4 {8} variable struct stack local lval vreg (7.14.4)
PTR "ptr" 7.14.4 {0} 
IDENT "stack" 7.14.15 {0} 
IDENT "stack" 7.14.22 {0} 
ALLOC "alloc" 7.14.30 {8} struct stack vreg 
IDENT "stack" 7.14.43 {0} 
'=' "=" 7.15.16 {8} string array lval vreg vaddr 
ARROW "->" 7.15.9 {8} string array lval vaddr 
IDENT "stack" 7.15.4 {8} variable struct stack local lval (7.14.4)
IDENT "data" 7.15.11 {0} 
ALLOC "alloc" 7.15.18 {8} string array 
ARRAY "array" 7.15.24 {0} 
STRING "string" 7.15.30 {0} 
IDENT "size" 7.15.40 {8} int variable param lval (7.13.30)
'=' "=" 7.16.16 {8} int lval vreg vaddr 
ARROW "->" 7.16.9 {8} int lval vaddr 
IDENT "stack" 7.16.4 {8} variable struct stack local lval (7.14.4)
IDENT "size" 7.16.11 {0} 
IDENT "size" 7.16.18 {8} int variable param lval (7.13.30)
'=' "=" 7.17.15 {8} int lval vreg vaddr 
ARROW "->" 7.17.9 {8} int lval vaddr 
IDENT "stack" 7.17.4 {8} variable struct stack local lval (7.14.4)
IDENT "top" 7.17.11 {0} 
'-' "-" 7.17.18 {8} int const vreg 
INTCON "1" 7.17.19 {8} int const 
RETURN "return" 7.18.4 {8} 
IDENT "stack" 7.18.11 {8} variable struct stack local lval (7.14.4)
FUNCTION "" 7.21.1 {8} void function 
TYPE_ID "" 7.21.1 {0} 
VOID "void" 7.21.1 {0} 
IDENT "push" 7.21.6 {0} 
PARAM "(" 7.21.11 {0} 
TYPE_ID "" 7.21.12 {9} variable struct stack param lval 
PTR "ptr" 7.21.12 {0} 
IDENT "stack" 7.21.23 {0} 
IDENT "stack" 7.21.30 {0} 
TYPE_ID "" 7.21.37 {9} string variable param lval 
STRING "string" 7.21.37 {0} 
IDENT "str" 7.21.44 {0} 
BLOCK "{" 7.21.49 {9} 
BLOCK "{" 7.22.4 {9} 
IF "if" 7.22.5 {9} 
NOT "not" 7.22.9 {9} int vreg 
LT "<" 7.22.25 {9} int vreg 
ARROW "->" 7.22.19 {9} int lval vaddr 
IDENT "stack" 7.22.14 {9} variable struct stack param lval (7.21.12)
IDENT "top" 7.22.21 {0} 
'-' "-" 7.22.39 {9} int vreg 
ARROW "->" 7.22.32 {9} int lval vaddr 
IDENT "stack" 7.22.27 {9} variable struct stack param lval (7.21.12)
IDENT "size" 7.22.34 {0} 
INTCON "1" 7.22.41 {9} int const 
CALL "(" 7.22.50 {0} void function (6.22.1)
IDENT "fail" 7.22.45 {0} 
STRINGCON ""stack->top < stack->size - 1"" 7.22.51 {9} string const 
STRINGCON ""40-arraystack.oc"" 7.22.83 {9} string const 
INTCON "22" 7.22.103 {9} int const 
';' ";" 7.22.108 {0} 
'=' "=" 7.23.15 {9} int lval vreg vaddr 
ARROW "->" 7.23.9 {9} int lval vaddr 
IDENT "stack" 7.23.4 {9} variable struct stack param lval (7.21.12)
IDENT "top" 7.23.11 {0} 
'+' "+" 7.23.28 {9} int vreg 
ARROW "->" 7.23.22 {9} int lval vaddr 
IDENT "stack" 7.23.17 {9} variable struct stack param lval (7.21.12)
IDENT "top" 7.23.24 {0} 
INTCON "1" 7.23.30 {9} int const 
'=' "=" 7.24.28 {9} string lval vreg vaddr 
INDEX "[" 7.24.15 {9} string lval vaddr 
ARROW "->" 7.24.9 {9} string array lval vaddr 
IDENT "stack" 7.24.4 {9} variable struct stack param lval (7.21.12)
IDENT "data" 7.24.11 {0} 
ARROW "->" 7.24.21 {9} int lval vaddr 
IDENT "stack" 7.24.16 {9} variable struct stack param lval (7.21.12)
IDENT "top" 7.24.23 {0} 
IDENT "str" 7.24.30 {9} string variable param lval (7.21.37)
FUNCTION "" 7.27.1 {9} string function 
TYPE_ID "" 7.27.1 {0} 
STRING "string" 7.27.1 {0} 
IDENT "pop" 7.27.8 {0} 
PARAM "(" 7.27.12 {0} 
TYPE_ID "" 7.27.13 {10} variable struct stack param lval 
PTR "ptr" 7.27.13 {0} 
IDENT "stack" 7.27.24 {0} 
IDENT "stack" 7.27.31 {0} 
BLOCK "{" 7.27.38 {10} 
TYPE_ID "" 7.28.4 {10} string variable local lval vaddr (7.28.4)
STRING "string" 7.28.4 {0} 
IDENT "tmp" 7.28.11 {0} 
INDEX "[" 7.28.28 {10} string lval vaddr 
ARROW "->" 7.28.22 {10} string array lval vaddr 
IDENT "stack" 7.28.17 {10} variable struct stack param lval (7.27.13)
IDENT "data" 7.28.24 {0} 
ARROW "->" 7.28.34 {10} int lval vaddr 
IDENT "stack" 7.28.29 {10} variable struct stack param lval (7.27.13)
IDENT "top" 7.28.36 {0} 
BLOCK "{" 7.29.4 {10} 
IF "if" 7.29.5 {10} 
NOT "not" 7.29.9 {10} int vreg 
NE "!=" 7.29.25 {10} int vreg 
ARROW "->" 7.29.19 {10} int lval vaddr 
IDENT "stack" 7.29.14 {10} variable struct stack param lval (7.27.13)
IDENT "top" 7.29.21 {0} 
'-' "-" 7.29.29 {10} int const vreg 
INTCON "1" 7.29.30 {10} int const 
CALL "(" 7.29.40 {0} void function (6.22.1)
IDENT "fail" 7.29.35 {0} 
STRINGCON ""stack->top != EMPTY"" 7.29.41 {10} string const 
STRINGCON ""40-arraystack.oc"" 7.29.64 {10} string const 
INTCON "29" 7.29.84 {10} int const 
';' ";" 7.29.89 {0} 
'=' "=" 7.30.15 {10} int lval vreg vaddr 
ARROW "->" 7.30.9 {10} int lval vaddr 
IDENT "stack" 7.30.4 {10} variable struct stack param lval (7.27.13)
IDENT "top" 7.30.11 {0} 
'-' "-" 7.30.28 {10} int vreg 
ARROW "->" 7.30.22 {10} int lval vaddr 
IDENT "stack" 7.30.17 {10} variable struct stack param lval (7.27.13)
IDENT "top" 7.30.24 {0} 
INTCON "1" 7.30.30 {10} int const 
RETURN "return" 7.31.4 {10} 
IDENT "tmp" 7.31.11 {10} string variable local lval (7.28.4)
FUNCTION "" 7.34.1 {10} int function 
TYPE_ID "" 7.34.1 {0} 
INT "int" 7.34.1 {0} 
IDENT "empty" 7.34.5 {0} 
PARAM "(" 7.34.11 {0} 
TYPE_ID "" 7.34.12 {11} variable struct stack param lval 
PTR "ptr" 7.34.12 {0} 
IDENT "stack" 7.34.23 {0} 
IDENT "stack" 7.34.30 {0} 
BLOCK "{" 7.34.37 {11} 
RETURN "return" 7.35.4 {11} 
EQ "==" 7.35.22 {11} int vreg 
ARROW "->" 7.35.16 {11} int lval vaddr 
IDENT "stack" 7.35.11 {11} variable struct stack param lval (7.34.12)
IDENT "top" 7.35.18 {0} 
'-' "-" 7.35.26 {11} int const vreg 
INTCON "1" 7.35.27 {11} int const 
FUNCTION "" 7.38.1 {11} int function 
TYPE_ID "" 7.38.1 {0} 
INT "int" 7.38.1 {0} 
IDENT "main" 7.38.5 {0} 
PARAM "(" 7.38.10 {0} 
TYPE_ID "" 7.38.11 {12} int variable param lval 
INT "int" 7.38.11 {0} 
IDENT "argc" 7.38.15 {0} 
TYPE_ID "" 7.38.21 {12} string array variable param lval 
ARRAY "array" 7.38.21 {0} 
STRING "string" 7.38.27 {0} 
IDENT "argv" 7.38.35 {0} 
BLOCK "{" 7.38.41 {12} 
TYPE_ID "" 7.39.4 {12} variable struct stack local lval vreg (7.39.4)
PTR "ptr" 7.39.4 {0} 
IDENT "stack" 7.39.15 {0} 
IDENT "stack" 7.39.22 {0} 
CALL "(" 7.39.40 {7} function struct stack (7.13.1)
IDENT "new_stack" 7.39.30 {0} 
INTCON "100" 7.39.41 {12} int const 
TYPE_ID "" 7.40.4 {12} int variable local lval vreg (7.40.4)
INT "int" 7.40.4 {0} 
IDENT "argi" 7.40.8 {0} 
INTCON "0" 7.40.15 {12} int const 
WHILE "while" 7.41.4 {12} 
LT "<" 7.41.16 {12} int vreg 
IDENT "argi" 7.41.11 {12} int variable local lval (7.40.4)
IDENT "argc" 7.41.18 {12} int variable param lval (7.38.11)
BLOCK "{" 7.41.24 {12} 
CALL "(" 7.42.12 {8} void function (7.21.1)
IDENT "push" 7.42.7 {0} 
IDENT "stack" 7.42.13 {12} variable struct stack local lval (7.39.4)
INDEX "[" 7.42.24 {12} string lval vaddr 
IDENT "argv" 7.42.20 {12} string array variable param lval (7.38.21)
IDENT "argi" 7.42.25 {12} int variable local lval (7.40.4)
'=' "=" 7.43.12 {12} int variable local lval vreg 
IDENT "argi" 7.43.7 {12} int variable local lval (7.40.4)
'+' "+" 7.43.19 {12} int vreg 
IDENT "argi" 7.43.14 {12} int variable local lval (7.40.4)
INTCON "1" 7.43.21 {12} int const 
WHILE "while" 7.45.4 {12} 
NOT "not" 7.45.11 {12} int vreg 
CALL "(" 7.45.21 {10} int function (7.34.1)
IDENT "empty" 7.45.15 {0} 
IDENT "stack" 7.45.22 {12} variable struct stack local lval (7.39.4)
BLOCK "{" 7.45.30 {12} 
CALL "(" 7.46.14 {3} void function (6.26.1)
IDENT "putstr" 7.46.7 {0} 
CALL "(" 7.46.19 {9} string function (7.27.1)
IDENT "pop" 7.46.15 {0} 
IDENT "stack" 7.46.20 {12} variable struct stack local lval (7.39.4)
CALL "(" 7.47.14 {1} void function (6.24.1)
IDENT "putchr" 7.47.7 {0} 
CHARCON "'\n'" 7.47.15 {12} int const 
RETURN "return" 7.49.4 {12} 
INTCON "0" 7.49.11 {12} int const 
