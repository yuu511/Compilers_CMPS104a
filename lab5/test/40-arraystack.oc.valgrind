==24261== Memcheck, a memory error detector
==24261== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==24261== Using Valgrind-3.14.0 and LibVEX; rerun with -h for copyright info
==24261== Command: ../oc 40-arraystack.oc
==24261== 
==24261== 
==24261== HEAP SUMMARY:
==24261==     in use at exit: 0 bytes in 0 blocks
==24261==   total heap usage: 1,447 allocs, 1,447 frees, 204,975 bytes allocated
==24261== 
==24261== All heap blocks were freed -- no leaks are possible
==24261== 
==24261== For counts of detected and suppressed errors, rerun with: -v
==24261== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
