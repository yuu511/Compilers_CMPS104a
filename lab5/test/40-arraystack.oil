.struct stack
.field int top
.field ptr data
.field int size
.end 
new_stack: .function struct stack 
           .param int size
           .local struct stack stack
           stack = call malloc(sizeof struct stack)  
           $t0:p = size * sizeof ptr
           -> = call malloc($t0:p)  
           stack->stack.size = size  
           $t1:i = -1  
           stack->stack.top = $t1:i  
           return stack
           return
           .end
push:      .function 
           .param struct stack stack
           .param ptr str
.if0:      $t2:i = stack->stack.top  
           $t4:i = stack->stack.size  
           $t3:i = $t4:i - 1
           $t1:i = $t2:i < $t3:i
           $t0:i = not $t1:i  
           goto .fi0 if not $t0:i  
.th0:      call fail("stack->top < stack->size - 1","40-arraystack.oc",22)  
.fi0:
           $t6:i = stack->stack.top  
           $t5:i = $t6:i + 1
           stack->stack.top = $t5:i  
           $t7:p = stack->stack.data  
           $t8:i = stack->stack.top  
           $t7:p[$t8:i * :c] = str  
           return
           .end
pop:       .function ptr 
           .param struct stack stack
           .local ptr tmp
           $t1:p = stack->stack.data  
           $t2:i = stack->stack.top  
           $t0:c = $t1:p[$t2:i * :c]  
           tmp = $t0:c  
.if1:      $t5:i = stack->stack.top  
           $t6:i = -1  
           $t4:i = $t5:i != $t6:i
           $t3:i = not $t4:i  
           goto .fi1 if not $t3:i  
.th1:      call fail("stack->top != EMPTY","40-arraystack.oc",29)  
.fi1:
           $t8:i = stack->stack.top  
           $t7:i = $t8:i - 1
           stack->stack.top = $t7:i  
           return tmp
           return
           .end
empty:     .function int 
           .param struct stack stack
           $t1:i = stack->stack.top  
           $t2:i = -1  
           $t0:i = $t1:i == $t2:i
           return $t0:i
           return
           .end
main:      .function int 
           .param int argc
           .param ptr argv
           .local struct stack stack
           .local int argi
           $t0:p = call new_stack(100)  
           stack = $t0:p  
           argi = 0  
.wh0:      goto .od0 if argi > argc
.do0:      $t1:c = argv[argi * :c]  
           call push(stack,$t1:c)  
           $t2:i = argi + 1
           argi = $t2:i  
           goto .wh0  
.od0
.wh1:      $t4:i = call empty(stack)  
           $t3:i = not $t4:i  
           goto .od1 if not $t3:i  
.do1:      $t5:p = call pop(stack)  
           call putstr($t5:p)  
           call putchr('\n')  
           goto .wh1  
.od1
           return 0
           return
           .end
