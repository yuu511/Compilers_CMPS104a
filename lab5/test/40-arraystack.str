string_set[   0]:    7296690267450279874 0x560542b21d58->"chr"
                   13348923631108115385 0x560542b22718->"putstr"
string_set[   5]:    5111042995199451563 0x560542b26738->""stack->top < stack->size - 1""
                   12679825168387489523 0x560542b22278->"num"
string_set[   6]:    9317282556992397945 0x560542b25e88->"if"
string_set[   7]:    2297668033614959926 0x560542b2a838->"0"
                    6814028903552918164 0x560542b236d8->"array"
string_set[  10]:    1583576014053361350 0x560542b228d8->"str"
                   13497578857207247412 0x560542b21318->"string"
                   11597697714117577063 0x560542b244f8->"="
string_set[  11]:    1396094175042499165 0x560542b23798->"<"
string_set[  13]:   12868445110721718657 0x560542b21978->")"
string_set[  16]:   10159970873491820195 0x560542b25478->"1"
                   11911796782241914524 0x560542b20088->"void"
string_set[  18]:   13581035385482130682 0x560542b270b8->"["
string_set[  19]:    5344662657899890615 0x560542b21a38->";"
string_set[  20]:   11868120863286193964 0x560542b29b88->"main"
                   15816506932648033945 0x560542b21258->"("
                    2497302301362042119 0x560542b23fd8->"new_stack"
string_set[  21]:    2617056156435215158 0x560542b28498->"!="
string_set[  23]:    5266873771696973564 0x560542b2a6f8->"argi"
                    6340238197210063670 0x560542b2a528->"100"
                   14752253762927134915 0x560542b23918->"data"
string_set[  24]:   13934964484049229816 0x560542b23a38->"size"
string_set[  25]:   11166701219533395393 0x560542b22ef8->"getstr"
                    5320480286546035437 0x560542b23558->"stack"
string_set[  28]:   13945322250941355849 0x560542b23498->"struct"
string_set[  29]:    1370730858159036685 0x560542b26d88->"+"
                   14635384904416252381 0x560542b25578->"return"
string_set[  34]:    7211130878751977441 0x560542b21b98->"putchr"
                    8290819294854170237 0x560542b24538->"alloc"
string_set[  35]:   14736352883882276958 0x560542b297d8->"=="
string_set[  37]:    6192799629375248351 0x560542b2ba68->"'\n'"
                   12291392672316406109 0x560542b21898->"line"
string_set[  39]:     549611050806166819 0x560542b27608->"pop"
                    8939588259793473717 0x560542b268b8->""40-arraystack.oc""
                   18413295556186398326 0x560542b220b8->"putint"
string_set[  40]:   10319632053171904911 0x560542b2a9c8->"while"
string_set[  42]:    1815486778388015849 0x560542b257e8->"push"
string_set[  43]:    7372855250431704371 0x560542b253b8->"-"
string_set[  44]:      21606623943142546 0x560542b23618->"{"
string_set[  45]:    6142509188972423790 0x560542b21498->""
string_set[  47]:   13686721134187162920 0x560542b23d58->"}"
string_set[  51]:    3701972582333163920 0x560542b29d48->"argc"
string_set[  52]:    8832308542017959627 0x560542b269a8->"22"
string_set[  53]:   12051031859636400188 0x560542b27318->"]"
string_set[  57]:    9386170080689500517 0x560542b213d8->"expr"
string_set[  58]:   13653664462152793141 0x560542b289e8->"29"
                    4017852018407011111 0x560542b21658->"file"
                    3729804957429652673 0x560542b1ad08->"<<ROOT>>"
string_set[  59]:   16579916199511856337 0x560542b231f8->"getln"
string_set[  61]:   14643508047410943861 0x560542b2a0a8->"argv"
                   14849783028996024063 0x560542b25fc8->"not"
string_set[  63]:    9611428142580219387 0x560542b21158->"fail"
string_set[  64]:   12597418120160561823 0x560542b217d8->"int"
string_set[  65]:   16247530808313002034 0x560542b23c18->"top"
string_set[  70]:   12099763674597357221 0x560542b287b8->""stack->top != EMPTY""
                   14805121035671447500 0x560542b24908->"->"
                    9201800968376702500 0x560542b22bf8->"getchr"
string_set[  71]:   14526079891305879776 0x560542b238d8->">"
string_set[  73]:     232592375681467293 0x560542b27ae8->"tmp"
string_set[  75]:     370180285156928338 0x560542b29138->"empty"
string_set[  76]:   17176659130743002837 0x560542b23e18->"ptr"
string_set[  78]:    4706206694200543916 0x560542b21198->","
load_factor = 0.810
bucket_count = 79
max_bucket_size = 3
