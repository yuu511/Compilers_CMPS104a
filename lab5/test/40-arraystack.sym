stack (7.7.1) {0} struct stack 
   data (7.8.4) string array field 0
   size (7.9.4) int field 1
   top (7.10.4) int field 2

push (7.21.1) {8} void function 
   stack (7.21.12) {9} variable struct stack param lval 0
   str (7.21.37) {9} string variable param lval 1

empty (7.34.1) {10} int function 
   stack (7.34.12) {11} variable struct stack param lval 0

new_stack (7.13.1) {7} function struct stack 
   size (7.13.30) {8} int variable param lval 0
   stack (7.14.4) {8} variable struct stack local lval 1

pop (7.27.1) {9} string function 
   stack (7.27.13) {10} variable struct stack param lval 0
   tmp (7.28.4) {10} string variable local lval 1

getln (6.30.1) {6} string function 

putchr (6.24.1) {1} void function 

getchr (6.28.1) {4} int function 

fail (6.22.1) {0} void function 

getstr (6.29.1) {5} string function 

putint (6.25.1) {2} void function 

main (7.38.1) {11} int function 
   argc (7.38.11) {12} int variable param lval 0
   argv (7.38.21) {12} string array variable param lval 1
   stack (7.39.4) {12} variable struct stack local lval 2
   argi (7.40.4) {12} int variable local lval 3

putstr (6.26.1) {3} void function 

