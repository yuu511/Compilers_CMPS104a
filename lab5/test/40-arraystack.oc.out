CONDITIONAL IF STATEMENT SYMBOL TOK_NOT 
CONDITIONAL IF STATEMENT SYMBOL TOK_NOT 
CONDITIONAL WHILE STATEMENT SYMBOL TOK_LT 
CONDITIONAL WHILE STATEMENT SYMBOL TOK_NOT 
